File: scripts/generate-brands.js
#!/usr/bin/env node
/**
 * Scans static/images/brand_logo/ and generates data/brands.json
 * Run: node scripts/generate-brands.js
 * Add new logo files to the folder, run this script, then refresh the page.
 */

const fs = require('fs');
const path = require('path');

const BRAND_LOGO_DIR = path.join(__dirname, '..', 'static', 'images', 'brand_logo');
const OUTPUT_FILE = path.join(__dirname, '..', 'data', 'brands.json');
const IMAGE_EXTENSIONS = new Set(['.png', '.jpg', '.jpeg', '.svg', '.webp', '.gif']);

function filenameToName(filename) {
  const name = path.basename(filename, path.extname(filename));
  return name
    .replace(/[-_]/g, ' ')
    .replace(/\b\w/g, c => c.toUpperCase());
}

try {
  const files = fs.readdirSync(BRAND_LOGO_DIR);
  const brands = files
    .filter(file => {
      const ext = path.extname(file).toLowerCase();
      return IMAGE_EXTENSIONS.has(ext);
    })
    .sort()
    .map(file => ({
      name: filenameToName(file),
      logo: `static/images/brand_logo/${file}`
    }));

  const output = { brands };
  fs.writeFileSync(OUTPUT_FILE, JSON.stringify(output, null, 2));
  console.log(`Generated brands.json with ${brands.length} logos`);
} catch (err) {
  console.error('Error:', err.message);
  process.exit(1);
}

File: scripts/generate-brands.py
#!/usr/bin/env python3
"""
Scans static/images/brand_logo/ and generates data/brands.json
Run: python3 scripts/generate-brands.py
Add new logo files to the folder, run this script, then refresh the page.
"""

import os
import json

BRAND_LOGO_DIR = os.path.join(os.path.dirname(__file__), '..', 'static', 'images', 'brand_logo')
OUTPUT_FILE = os.path.join(os.path.dirname(__file__), '..', 'data', 'brands.json')
IMAGE_EXTENSIONS = {'.png', '.jpg', '.jpeg', '.svg', '.webp', '.gif'}


def filename_to_name(filename):
    name = os.path.splitext(filename)[0]
    return name.replace('-', ' ').replace('_', ' ').title()


def main():
    files = sorted(os.listdir(BRAND_LOGO_DIR))
    brands = []
    for f in files:
        ext = os.path.splitext(f)[1].lower()
        if ext in IMAGE_EXTENSIONS:
            brands.append({
                "name": filename_to_name(f),
                "logo": f"static/images/brand_logo/{f}"
            })

    with open(OUTPUT_FILE, 'w') as out:
        json.dump({"brands": brands}, out, indent=2)

    print(f"Generated brands.json with {len(brands)} logos")


if __name__ == '__main__':
    main()
